README.md

```diff
@@ -1,2 +1,422 @@
+
+```swift
+@Instantiable
+public struct MyApp: App {
+    public var body: some Scene {
+        WindowGroup {
+            if let user = userService.user {
+                // Returns a new instance of a `LoggedInContentView`.
+                loggedInContentViewInstantiator.instantiate(user)
+            } else {
+                // Returns a new instance of a `LoggedOutContentView`.
+                loggedOutContentViewInstantiator.instantiate()
+            }
+        }
+    }
+
+    public init(loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>, loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>, userService: UserService) {
+        self.loggedInContentViewInstantiator = loggedInContentViewInstantiator
+        self.loggedOutContentViewInstantiator = loggedOutContentViewInstantiator
+        self.userService = userService
+    }
+
+    /// A private property that knows how to instantiate a logged-in content view.
+    @Instantiated
+    private let loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>
+
+    /// A private property that knows how to instantiate a logged-out content view.
+    @Instantiated
+    private let loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>
+
+    @ObservedObject
+    @Instantiated
+    private var userService: UserService
+}
+```
+
+It is possible to write a `ForwardingInstantiator` with a type-erased second generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+### Creating the root of your dependency tree
+
+SafeDI automatically finds the root(s) of your dependency tree, and creates an extension on each root that contains a `public init()` method that instantiates the dependency tree.
+
+An `@Instantiable` type qualifies as the root of a dependency tree if and only if:
+
+1. The type‘s SafeDI-injected properties are all `@Instantiated`
+2. The type is not instantiated by another `@Instantiable` type
+
+### Comparing SafeDI and Manual Injection: Key Differences
+
+SafeDI is designed to be simple to adopt and minimize architectural changes required to get the benefits of a compile-time safe DI system. Despite this design goal, there are a few key differences between projects that utilize SafeDI and projects that don‘t. As the benefits of this system are clearly outlined in the [Features](#features) section above, this section outlines the pattern changes required to utilize a DI system like SafeDI.
+
+#### Instantiating objects
+
+In a manual DI system, it is common to directly call your dependencies‘ `init(…)` methods. When utilizing SafeDI, you must rely on `@Instantiated`-decorated properties to instantiate your dependencies for you. Calling a dependency‘s `init(…)` method directly effectively exits the SafeDI-built dependency tree.
```

Comment:
> When utilizing SafeDI, you must rely on `@Instantiated`-decorated properties to instantiate your dependencies for you.
What happens if you don't? Say you start passing around a `@Instantiated` dependency your class received, will this cause any issue?

Reply:
You can call `init` yourself, but then you're basically exiting the SafeDI dependency tree and there are no guarantees re when something is instantiated. Sounds like I should make that more clear here 🙂

Reply:
If someone forgets to add `@Instantiated` to a property, is this something they will notice easily?

Reply:
short answer: yes they will notice quickly and easily.

If they fail add `@Instantiated`, `@Received`, or `@Forwarded` to a property on an `@Instantiatable` type, then SafeDI will fail the build (with a FixIt) until they write an `init` method that does not include this property as an argument. If they accept the FixIt, they'll need to instantiate the property themselves in `init(...)`, which should be a sign that they're not using DI for this property.

Reply:
I've now answered Cyril's q in this section!